### ISSUE TITLE
Creating a set of river plots for the paper

### ISSUE THREAD

**Issue by ericagol** [langfzac/Photodynamics.jl#26]
- [ ] Write a routine for computing river plots for the flux, as well as for the various derivatives.
- [ ] Create some plots for the paper showing various examples of river plots.

**Comment by ericagol** [comment]
@langfzac  If I want to compute an array with the derivatives of the flux with respect to each of the mass ratios, radius-ratios  initial orbital elements, limb-darkening & stellar density, what is the procedure for doing this?

**Comment by langfzac** [comment]
Do you mean in general? Or in the inference.jl file?

**Comment by langfzac** [comment]
In general, the flux derivatives are contained in the `Lightcurve` after running the necessary code to compute them. Run the example in README with `grad=true`, and add the number of nbody parameters (`7*n`) as an argument to the `Lightcurve` constructor. The rest is the same. 

The fields are defined as below:

https://github.com/langfzac/Photodynamics.jl/blob/a7be91df1881dbbfc13f26e23bc3d1b979a52968/src/Lightcurve.jl#L3-L18

**Comment by ericagol** [comment]
> Do you mean in general? Or in the inference.jl file?

I meant in inference.jl, but the general case is fine too.

**Comment by ericagol** [comment]
> 

It looks like this contains the cartesian coordinate derivatives, not the initial orbital elements.

**Comment by ericagol** [comment]
> In general, the flux derivatives are contained in the `Lightcurve` after running the necessary code to compute them. Run the example in README with `grad=true`, and add the number of nbody parameters (`7*n`) as an argument to the `Lightcurve` constructor. The rest is the same.
> 
> The fields are defined as below:
> 
> https://github.com/langfzac/Photodynamics.jl/blob/a7be91df1881dbbfc13f26e23bc3d1b979a52968/src/Lightcurve.jl#L3-L18

It looks like this contains the initial cartesian coordinate derivatives, not the initial orbital element derivatives.

**Comment by langfzac** [comment]
> > 
> 
> It looks like this contains the cartesian coordinate derivatives, not the initial orbital elements.

Yes, I haven't included those. Right now, you'd have to do the transformations manually. (There's an example in inference.jl -- see `grad_compute_photometry`)

**Comment by ericagol** [comment]
Okay, that gets back to my question about what gets returned by that routine.

**Comment by langfzac** [comment]
Oh right, it returns the flux and jacobian array. Although, that function in inference.jl is optimized for that particular problem/system. There's no generalized function that computes the flux (and/or jacobian) for an arbitrary parameter vector.

**Comment by ericagol** [comment]
@langfzac  I still haven't been able to figure this out, so I could use some help with this.

**Comment by ericagol** [comment]
@langfzac  Thanks to your help, I now have some riverplots of derivatives working.  Here is the plot with respect to the radius of the star.  The blue is *negative*, which I think is indicating that the radius of the star increases, then the duration increases, so in ingress & egress the flux becomes smaller (i.e. the transit widens).  It's hard to see, but the flux is also negative in the middle of the transits, but goes to zero mid-transit.   This confused me at first, but I think this is due to the fact that the radius-ratio is being held constant, so the depth of transit is the same, but the stretching of the transit due to a longer duration means that the transit needs to get slightly deeper.

![dfdrstar](https://user-images.githubusercontent.com/243664/194960668-6968f662-484d-4235-ac89-a640393d13da.png)

**Comment by ericagol** [comment]
@langfzac  Quick point about the derivative with respect to the mass of the star.   I find this to be identically zero, which surprised me at first. I think this may be because we give the periods and times as the initial elements, and so if the mass of the star changes, for the period to remain the same, the semi-major axes need to change to keep the period & initial transit times constant.  So, there is no dependence upon the mass of the star.

**Comment by ericagol** [comment]
I'm confused about some of the derivatives with respect to the dynamical parameters, which aren't behaving as I would have expected....   but, the model is complicated...
